nova/tests/unit/scheduler/weights/test_weights_resize_same_shard.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2022 SAP
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+Tests for scheduler prefer-resize-to-same-host weigher.
+"""
+from nova import objects
+from nova.scheduler import weights
+from nova.scheduler.weights import resize_same_shard as same_shard
+from nova import test
+from nova.tests.unit.scheduler import fakes
+from nova.tests import uuidsentinel as uuids
+
+
+class PreferSameShardOnResizeWeigherTestCase(test.NoDBTestCase):
+    def setUp(self):
+        super(PreferSameShardOnResizeWeigherTestCase, self).setUp()
+        self.weight_handler = weights.HostWeightHandler()
+        self.weighers = [same_shard.PreferSameShardOnResizeWeigher()]
+
+        self.instance1 = objects.Instance(host='host1', uuid=uuids.instance_1)
+        self.instance2 = objects.Instance(host='host1', uuid=uuids.instance_2)
+        self.instance3 = objects.Instance(host='host2', uuid=uuids.instance_3)
+        self.instance4 = objects.Instance(host='host3', uuid=uuids.instance_4)
+        self.aggs1 = [objects.Aggregate(id=1, name='az-a', hosts=['host1',
+                                                                  'host3']),
+                      objects.Aggregate(id=1, name='vc-a-0', hosts=['host1',
+                                                                    'host3'])]
+        self.aggs2 = [objects.Aggregate(id=2, name='az-b', hosts=['host2']),
```

Comment:
Why are these hosts in different `az` aggregates? Reading the tests, you might as well test a weigher that's based on the az aggregates and get the same result. Additiionally, resizes across AZs are not supported afaik, so this is a bad example. Just put them in the same az.